src/Aspire.Hosting.Keycloak/KeycloakResouceBuilderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Keycloak;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Keycloak resources to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class KeycloakResouceBuilderExtensions
+{
+    private const string AdminEnvVarName = "KEYCLOAK_ADMIN";
+    private const string AdminPasswordEnvVarName = "KEYCLOAK_ADMIN_PASSWORD";
+    private const string DefaultAdmin = "admin";
+    private const string DefaultAdminPassword = "admin";
+    private const int DefaultContainerPort = 8080;
+
+    /// <summary>
+    /// Adds a Keycloak container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. </param>
+    /// <param name="port">The host port that the underlying container is bound to when running locally.</param>
+    /// <param name="admin">The parameter used as the admin for the Keycloak resource. If <see langword="null"/> a default value will be used.</param>
+    /// <param name="adminPassword">The parameter used as the admin password for the Keycloak resource. If <see langword="null"/> a default password will be used.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<KeycloakResource> AddKeycloak(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int? port = null,
+        IResourceBuilder<ParameterResource>? admin = null,
+        IResourceBuilder<ParameterResource>? adminPassword = null)
+    {
+        var resource = new KeycloakResource(name);
+
+        var keycloak = builder
+            .AddResource(resource)
+            .WithImage(KeycloakContainerImageTags.Image)
+            .WithImageRegistry(KeycloakContainerImageTags.Registry)
+            .WithImageTag(KeycloakContainerImageTags.Tag)
+            .WithHttpEndpoint(port: port, targetPort: DefaultContainerPort)
+            .WithEnvironment(AdminEnvVarName, admin?.Resource.Value ?? DefaultAdmin)
+            .WithEnvironment(AdminPasswordEnvVarName, adminPassword?.Resource.Value ?? DefaultAdminPassword);
```

Comment:
This should be deferred using string interpolation or the ReferenceExpression. See https://github.com/dotnet/aspire/blob/7f034253814d75622c87e39dadd6eaaaab13a7f9/src/Aspire.Hosting.PostgreSQL/PostgresServerResource.cs#L39-L42

https://github.com/dotnet/aspire/blob/7f034253814d75622c87e39dadd6eaaaab13a7f9/src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs#L45-L46

Also see default password generation:

https://github.com/dotnet/aspire/blob/7f034253814d75622c87e39dadd6eaaaab13a7f9/src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs#L34

Reply:
Done!